cmake/CMakeLists.txt

```diff
@@ -1816,14 +1816,11 @@ if (onnxruntime_USE_CUDA)
   set(CMAKE_CUDA_RUNTIME_LIBRARY Shared)
   enable_language(CUDA)
   message( STATUS "CMAKE_CUDA_COMPILER_VERSION: ${CMAKE_CUDA_COMPILER_VERSION}")
+  set(CMAKE_CUDA_STANDARD 17)
   if (WIN32)
-    set(CMAKE_CUDA_STANDARD 17)
     foreach(CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORY ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
```

Comment:
why do you make this change?

Reply:
Otherwise it would get error like:
/onnxruntime_src/onnxruntime/core/providers/cpu/tensor/upsample.h:614:4: error: ‘if constexpr’ only available with ‘-std=c++17’ or ‘-std=gnu++17’ [-Werror]
(https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=665920&view=logs&j=7536d2cd-87d4-54fe-4891-bfbbf2741d83&t=14ed71ea-dac1-51ae-e9ca-3f24cf8795d8)

Reply:
For this error, i think you can separate UpsampleBase from upsample.h

[https://github.com/microsoft/onnxruntime/blob/7e092a7e3f8f9f590de0ac5548035e8fa4d512f8/onnxruntime/core/providers/cuda/tensor/upsample.h#L8](https://github.com/microsoft/onnxruntime/blob/7e092a7e3f8f9f590de0ac5548035e8fa4d512f8/onnxruntime/core/providers/cuda/tensor/upsample.h#L8)

Reply:
Fix by separating UpsampleBase from upsample.h

Reply:
Fix by separating UpsampleBase from upsample.h